contest/views.py

```diff
@@ -18,64 +18,140 @@
     SOFTWARE.
     '''
 
-from django.http import HttpResponse
+from django.http import HttpResponseRedirect
 from django.shortcuts import render,get_object_or_404
 from datetime import datetime
 from index.views import custom_proc
 from django.template import RequestContext
+from django.forms.models import model_to_dict
 
-def index(request):
-    
-    problem1 = {'id':1,'name':'First Problem','ppl_pass':2,'ppl_not_pass':4}
-    problem2 = {'id':2,'name':'Second Problem','ppl_pass':1,'ppl_not_pass':8}
-    problem3 = {'id':3,'name':'third Problem','ppl_pass':5,'ppl_not_pass':1}
-    problemList = [problem1,problem2,problem3]
-
-    ppl1 = {'name':'naruto'}
-    ppl2 = {'name':'obama'}
-    
-    contestantList = [ppl1,ppl2]
+from contest.models import Contest
+from contest.models import Contestant
+from contest.models import Clarification
 
-    ppl3 = {'name':'forest'}
-    ppl4 = {'name':'gump'}
+from contest.forms import ContestForm
 
-    coownerList = [ppl3,ppl4]
+from problem.models import Submission
+from problem.models import Problem
 
-    contest1 = {'id':1,'name':'First contest','start_time':'2014/11/29 10:11:22','end_time':'2014/12/05 02:12:50',
-                'contestant_number':200,'owner':'ma in joe','problem_list':problemList,'contestant_list':contestantList,
-                'coowner_list':coownerList}
+from general_tools.log import get_logger
 
-    contestList = [contest1]
+def archive(request):
+    #to store contest basic info and contestants
+    contest_list = []
+    #store contest basic info only
+    contest_info_list = Contest.objects.order_by('-start_time')
+    for contest in contest_info_list:
+        contestants = Contestant.objects.filter(contest = contest)
+        contest_list.append({'contest':contest,'contestants':contestants})
 
-    #contest_list = Contest.objects.order_by('-contest_id')
-    #problem_list = Problem.objects.all()
-    #contestant_list = {1:'naruto',2:'obama'}
-    #coowner_list = {1:'teemo',2:'lux',3:'jinx'}
-    return render(request, 'contest/contestArchive.html',{'contest_list':contestList,'admin':1},
-                context_instance = RequestContext(request, processors = [custom_proc]))
+    return render(request, 'contest/contestArchive.html',
+        {'contest_list':contest_list,'admin':1},
+        context_instance = RequestContext(request, processors = [custom_proc]))
 
 def contest(request,contest_id):
 
     serverTime = datetime.now().strftime('%Y/%m/%d %H:%M:%S')
-
-    problem1 = {'id':1,'name':'First Problem','ppl_pass':2,'ppl_not_pass':4}
-    problem2 = {'id':2,'name':'Second Problem','ppl_pass':1,'ppl_not_pass':8}
-    problem3 = {'id':3,'name':'third Problem','ppl_pass':5,'ppl_not_pass':1}
-    problemList = [problem1,problem2,problem3]
-
-    ppl1 = {'name':'naruto','score':['0/3','1/2','1/4'],'solved':'2','panalty':'833'}
-    ppl2 = {'name':'obama','score':['0/2','1/5','1/3'],'solved':'2','panalty':'83'}
+    contest = get_object_or_404(Contest,id = contest_id)
```

Comment:
I think it's better to log when users query a non-existing contest.
I'm not sure whether there will be any situation that cause it. If so, the log is convenience for us to trace the problem.